beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1186,7 +1186,19 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         &self,
         block_root: &Hash256,
     ) -> Result<Option<BlobSidecarList<T::EthSpec>>, Error> {
-        Ok(self.store.get_blobs(block_root)?)
+
+        let blobs = self.store.get_blobs(block_root)?;
+        match Some(blobs) {
+            Some(blobs) => Ok(blobs),
+            None => {
+                // if there are no blobs, but a block exists return an empty list
+                if let Some(_) = self.store.try_get_full_block(block_root)? {
```

Comment:
This check has already been performed when obtaining the `block_root`, and it returns a rejection (`404`) if block does not exist:

https://github.com/sigp/lighthouse/blob/f1f04bc68a5506c839693d25bc8c626c48ec8669/beacon_node/http_api/src/block_id.rs#L82-L92

https://github.com/sigp/lighthouse/blob/f1f04bc68a5506c839693d25bc8c626c48ec8669/beacon_node/http_api/src/block_id.rs#L116-L120

So I believe we could just supply `BlobSidecarList::default()` as default if `get_blobs` returns `None`, and even remove the `Option` from method signature and just return `Result<BlobSidecarList<T::EthSpec>, Error>`?

Reply:
Thanks for the feedback! I did not see that check! I've implemented what you recommended